watcher.go

```diff
@@ -254,6 +254,72 @@ func (w *Watcher) Wait(ctx context.Context) error {
 	}
 }
 
+// Watch will continuously watch for data updates and send template IDs to
+// the provided channel which templates have changes to be rendered. Useful
+// for when caller wants to process templates asynchronously. Only one Watch
+// should be called at given time and should not be called with Wait.
+func (w *Watcher) Watch(ctx context.Context, tmplCh chan string) error {
+	w.stopCh.drain()
+	go func() { w.waitingCh <- struct{}{} }()
```

Comment:
As waitingCh is currently only used in tests, non-testing usage of this will leak a Go routine each call as nothing will ever read from that channel.

Reply:
Just realized this doesn't appear in the tests below so I it could just be removed. In Wait() it is only used for testing.